Add documentation about tabtests (#106)
xodio/xod-docs: docs/reference/tabtests/README.md

---
title: Tabtest Reference

> version: 0.0.1

Reviewer: ```suggestion
version: 1.0.0
```

---

# Tabtest Reference

> Each tabtest are represented as a table, where:

Reviewer: ```suggestion
Each tabtest is represented as a table, where:
```

> - the first row contains columns with pin labels or [time flow](#time-flow) special column name

Reviewer: ```suggestion
- the first row contains pin labels or the special [time flow](#time-flow) name
```

> - all next rows contains columns with some input and expected output values. See [literals](#literals) section for details.

Reviewer: ```suggestion
- all subsequent rows contain cells with some input and expected output values. See [literals](#literals) section for details.
```

> - each row is one test case that executed within one [transaction](/docs/guide/execution-model/) of XOD program

Reviewer: ```suggestion
- each row is one test case that executes within one [transaction](/docs/guide/execution-model/) of a XOD program
```

> - BUT if tested node contains some state inside it [will be shared](/docs/guide/testing-patches/#impureness) between all test cases

Reviewer: ```suggestion
- if the tested node contains some state inside it [will be shared](/docs/guide/testing-patches/#impureness) between all test cases
```

<a name="literals"></a>

> ## Literals

Reviewer: No need for an anchor. It is created automatically.

Generally, literals for tabtests are the same that are used in the Inspector. But also there are few special literals for tabtests only.

> - **Numbers**

Reviewer: Bulleted lists are for short items only. Suggesting demoting the list to a series of paragraphs titled with H3 (three ###) headings

  <table class="ui table compact">

>     <tbody class="top aligned">

Reviewer: Suggest adding the title row and swapping two first columns.

Apply the same style to other types
